sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -310,6 +310,17 @@ class QueryCompilationErrorsSuite extends QueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("SECOND_FUNCTION_ARGUMENT_NOT_INTEGER: " +
+    "the second argument of 'date_add' function needs to be an integer") {
+    val e = intercept[AnalysisException] {
+      sql("select date_add('1982-08-15', 'x')").collect()
+    }
+    assert(e.getErrorClass === "SECOND_FUNCTION_ARGUMENT_NOT_INTEGER")
+    assert(e.getSqlState === "22023")
+    assert(e.getMessage.matches(
+      "The second argument of 'date_add' function needs to be an integer."))
```

Comment:
Let's just compare the error message directly:
```suggestion
    assert(e.getMessage ===
      "The second argument of 'date_add' function needs to be an integer.")
```